src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (!target.hidden && target instanceof Herbivore) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(target), 1);
```

Comment:
Consider using `filter` to create a new array without the dead animal instead of splicing the existing array. This approach avoids mutating the array during iteration, which can lead to errors and is generally considered a bad practice.